tests/console/commands/env/test_activate.py

```diff
@@ -77,6 +77,118 @@ def test_env_activate_prints_correct_script_on_windows(
     assert line == f'{prefix}"{tmp_venv.bin_dir / ext!s}"'
 
 
+@pytest.mark.parametrize(
+    "shell, ext, expected_prefix",
+    (
+        ("cmd", "activate.bat", ""),
+        ("pwsh", "activate.ps1", "& "),
+        ("powershell", "activate.ps1", "& "),
+    ),
+)
+@pytest.mark.skipif(not WINDOWS, reason="Only Windows shells")
+def test_env_activate_windows_shells_get_quoted_path_only(
+    tmp_venv: VirtualEnv,
+    mocker: MockerFixture,
+    tester: CommandTester,
+    shell: str,
+    ext: str,
+    expected_prefix: str,
+) -> None:
+    mocker.patch("shellingham.detect_shell", return_value=(shell, None))
+    mocker.patch("poetry.utils.env.EnvManager.get", return_value=tmp_venv)
+
+    tester.execute()
+
+    line = tester.io.fetch_output().rstrip("\n")
+    expected = f'{expected_prefix}"{tmp_venv.bin_dir / ext!s}"'
+    assert line == expected
+
+
+@pytest.mark.parametrize(
+    "shell, command, ext",
+    (
+        ("bash", "source", ""),
+        ("zsh", "source", ""),
+        ("fish", "source", ".fish"),
+        ("nu", "overlay use", ".nu"),
+        pytest.param(
+            "csh",
+            "source",
+            ".csh",
+            marks=pytest.mark.skipif(
+                WINDOWS, reason="csh activator not created on Windows"
+            ),
+        ),
+        pytest.param(
+            "tcsh",
+            "source",
+            ".csh",
+            marks=pytest.mark.skipif(
+                WINDOWS, reason="tcsh activator not created on Windows"
+            ),
+        ),
+        ("sh", "source", ""),
+    ),
+)
+def test_env_activate_unix_shells_get_command_with_path(
```

Comment:
This looks like a copy `test_env_activate_prints_correct_script` that also works for Windows. Can we just edit the original test instead of creating a new one?